qiskit/compiler/transpile.py

```diff
@@ -418,13 +419,17 @@ def _parse_transpile_args(circuits, backend,
     layout_method = _parse_layout_method(layout_method, num_circuits)
     routing_method = _parse_routing_method(routing_method, num_circuits)
     translation_method = _parse_translation_method(translation_method, num_circuits)
-    durations = _parse_instruction_durations(backend, instruction_durations, dt, circuits)
-    scheduling_method = _parse_scheduling_method(scheduling_method, num_circuits)
     seed_transpiler = _parse_seed_transpiler(seed_transpiler, num_circuits)
     optimization_level = _parse_optimization_level(optimization_level, num_circuits)
     output_name = _parse_output_name(output_name, circuits)
     callback = _parse_callback(callback, num_circuits)
 
+    durations = _parse_instruction_durations(backend, instruction_durations, dt, circuits)
+    scheduling_method = _parse_scheduling_method(scheduling_method, circuits)
+    if scheduling_method and not durations:
+        raise TranspilerError("Transpiling a circuit with a scheduling method or with delay "
+                              "instructions requires a backend or instruction_durations.")
```

Comment:
Just note that these lines make impossible to transpile circuits with only delays. barriers, and cal gates for simulator backend, while `test_transpile_delay_circuit_without_scheduling_method_or_durs` succeeds without these lines. However, I cannot think of any use case of simulating such circuits and these lines help provide more informative error message. So adding these lines makes sense to me.

Reply:
cal gates will populate durations, right? Is there a better way to check this? If I run `transpile(qc)` on a circuit that has delays, I get an error something like "Error: Duration for operation 'cx' on qubits [0, 1] was not found"

Reply:
I don't have any good idea; we may accept less informative error (no duration found), or, we may have to check if circuits are composed only by instructions which can populate durations (I think it's too much). I think we can leave these lines as you've added for now.